mate-panel/panel-run-dialog.c

```diff
@@ -326,17 +326,34 @@ panel_run_dialog_set_icon (PanelRunDialog *dialog,
 	}
 }
 
+static gchar *replace_command_string (const char *cmd)
+{
+	g_autofree gchar *path = NULL;
+	GRegex *regex = NULL;
+	char   *new_cmd = NULL;
+
+	regex = g_regex_new ("~/", 0, 0, NULL);
+	path = g_build_filename (g_get_home_dir (), "/", NULL);
+
+	new_cmd = g_regex_replace_literal (regex, cmd, -1, 0, path, 0, NULL);
+
+	g_regex_unref (regex);
+	return new_cmd;
+}
+
 static gboolean
 command_is_executable (const char   *command,
 		       int          *argcp,
 		       char       ***argvp)
 {
 	gboolean   result;
+	g_autofree gchar *new_command = NULL;
 	char     **argv;
 	char      *path;
 	int        argc;
 
-	result = g_shell_parse_argv (command, &argc, &argv, NULL);
+	new_command = replace_command_string (command);
```

Comment:
Instead of replacing the entire command, I would iterate through argv[1..n] and use the `^~/` regex. Something like:
```
for (int i = 1; i < g_strv_length (argv); i++) {
  regex = g_regex_new ("^~/", 0, 0, NULL);
  path = g_build_filename (g_get_home_dir (), "/", NULL);
  argv[i] = g_regex_replace_literal (regex, argv[i], -1, 0, path, 0, NULL);
}
```

Reply:
Good advice, but index can start from 0  ```int i = 0```
For example, run the following command
```~/pluma ~/some```